### DISCUSSION TITLE
Copy value from one control to another control

### DISCUSSION THREAD

**Discussion by shahmanish877** [elementor/elementor#15346] [Developers Community]
How can I copy a value of one control to another? I can't find anything similar to condition.

**Comment by ArekZw** [comment]
{{id_settings_to_copy.VALUE}}

**Comment by shahmanish877** [comment]
And what shall I use for condition so that it will copy or will it work without condition.
Also, can you help me on this 
https://github.com/elementor/elementor/issues/15309

**Comment by shahmanish877** [comment]
I think {{id_settings_to_copy.VALUE}} this works if we are build custom control. But I need value in Controls_Manager::TEXT when select2 value is selected.

**Comment by ArekZw** [comment]
I still don't understand what you want to achieve. Maybe it's about this?

```
$this->add_control(
   'xxxxxx',
   [
      'label' => __( 'XXXXXX', 'isb' ),
      'type' => Controls_Manager::SELECT2,
      'options'     => [
         'option1' => 'Option 1',
         'option2' => 'Option 2',
      ],
   ]
);


$this->add_control(
   'yyyyyy',
   [
      'label' => __( 'YYYYYY', 'isb' ),
      'type' => Controls_Manager::TEXT,
      'condition' => [
         'yyyyyy' => 'option1', // IF XXXXXX field == option1 this field will be displayed
      ],
      'selectors' => [
            '#something' => '{{xxxxxx.VALUE}} {{VALUE}} ',
      ],
   ]
);
```


2nd scenario: (display second field if first are have any value other than 'none':
```
$this->add_control(
   'xxxxxx',
   [
      'label' => __( 'XXXXXX', 'isb' ),
      'type' => Controls_Manager::SELECT2,
      'default' => 'none',
      'options'     => [
         'none' => 'none',
         'option1' => 'Option 1',
         'option2' => 'Option 2',
      ],
   ]
);


$this->add_control(
   'yyyyyy',
   [
      'label' => __( 'YYYYYY', 'isb' ),
      'type' => Controls_Manager::TEXT,
	'conditions' => [
		'terms' => [
			[
				'name' => 'xxxxxx',
				'operator' => '!==',
				'value' => 'none',
			],
		],
	],
      'selectors' => [
            '#something' => '{{xxxxxx.VALUE}} {{VALUE}} ',
      ],
   ]
);
```





(I wrote it from memory, it may have some little bug)

**Comment by shahmanish877** [comment]
```
              $this->add_control(
			'search_symbol',
			[
				'label' => __( 'Search Symbol', 'elementor-trading-view-widget' ),
				'type' => 'search_symbol',
				'default'=>'',
			]
		);

		$this->add_control(
		   'selected_symbol',
		   [
		      'label' => __( 'Selected Symbol', 'elementor-trading-view-widget' ),
		      'type' => Controls_Manager::TEXT,
		      'placeholder' => 'Selected Symbol will be displayed here',
		   ]
		);
```

This is my custom widget and I have tried to create a custom control (search_symbol) which works with select2 and generate a dynamic option with ajax API call. I can't figure out how to save the selected value to the editor and I have issued this topic (https://github.com/elementor/elementor/issues/15309) but no one replied to the issue so I went to try for an alternate solution where the selected option will be displayed to 'selected_symbol' when 'search_symbol' is selected. 
Hope you understand what I want.

**Reply by ArekZw** [reply]
Yes, now I understand.
To my knowledge, it cannot be done according to the idea of this topic. 

IMO #15309 is a better way to solve this problem, but I can't help you with that. Maybe someone has more experience creating custom controls.

**Comment by Aviad-Herman** [comment]
Hey guys,
Great conversation!
I'm converting this thread to a GitHub discussion so it will help others as well.
![image](https://user-images.githubusercontent.com/73067562/121927613-eeefd800-cd47-11eb-9fca-11272a1194f9.png)
